### PR TITLE
Add fluid merkle rewards claim connector

### PR DESCRIPTION
**Pull request by shriyatyagii**
- Adds `claim` and `claimOnBehalf` to claim rewards through Fluid reward distributor contract.

### CODE REVIEW

File: contracts/mainnet/connectors/merkleDistributor/main.sol [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+     * @param cycle_ Current epoch cycle.
+     * @param merkleProof_ Merkle proof that validates this claim.
+     * @param getId_ Id to get the cumulative amount.
+     */
+    function claim(
+        uint256 cumulativeAmount_,
+        address fToken_,
+        uint256 cycle_,
+        bytes32[] calldata merkleProof_,
+        uint256 getId_
```
Comment (thrilok209) [inline contracts/mainnet/connectors/merkleDistributor/main.sol:25]: @shriyatyagii We can completely remove getId i guess. Or we can add setId along with it? 

reward amount user is gonna get is not always eq to cumulativeAmount_.

Comment (shriyatyagii) [reply contracts/mainnet/connectors/merkleDistributor/main.sol:25]: Oh got it! Makes sense to add setId then

File: contracts/mainnet/connectors/merkleDistributor/main.sol [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+pragma solidity ^0.8.2;
+
+import {Basic} from "../../common/basic.sol";
+import {Events} from "./events.sol";
+import {IFluidMerkleDistributor} from "./interface.sol";
+import {TokenInterface} from "../../common/interfaces.sol";
+
+abstract contract FluidMerkle is Basic, Events {
+    IFluidMerkleDistributor internal constant MERKLE_DISTRIBUTOR =
+        IFluidMerkleDistributor(address(0));
```
Comment (thrilok209) [inline contracts/mainnet/connectors/merkleDistributor/main.sol:11]:
```suggestion
        IFluidMerkleDistributor(0x1d3D6d38Da24E9Ff3B17693f6971CAe62c3AA1DE);
```

**Comment by shriyatyagii** [comment]
Add connector -
`FLUID-INST-CLAIM-A`: [0xA846B6E912F03831137Ce3C0349D784F8d51cE78](https://etherscan.io/address/0xA846B6E912F03831137Ce3C0349D784F8d51cE78#code)

**Pull request merged**